Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMaterialQuantities.cs

```diff
@@ -22,11 +22,32 @@ public Objects.Other.MaterialQuantity MaterialQuantityToSpeckle(DB.Element eleme
     {
       if (material == null || element == null) 
         return null;
+      double volume = 0;
+      double area=0;
 
-      //Get quantities
-      double volume = element.GetMaterialVolume(material.Id);
-      double area = element.GetMaterialArea(material.Id, false); //To-Do: Do we need Paint-Materials
-
+      if(element is DB.MEPCurve )
+      {
+        // Area and Volume are computed based on geometry elements
+        GetGeometry(element, out List<DB.Mesh> meshes, out List<DB.Solid> solids);
+        volume = solids.Sum(s => s.Volume);
+        area = solids.Select(s => s.Faces.Cast<DB.Face>().Select(face => face.Area).Max()).Sum();
```

Comment:
Here it looks like you aren't filtering the solids to see if the MaterialElementId == materialId. Could this result in sending incorrect data? Assuming that the material is something that it isn't?

Reply:
Thanks for the very good question. I need to double-check this. I am not completely sure but I don’t see a problem as MEPCurves can only have 1 material. Also, I think that the MEPCurves do not have the material assigned on the solids. But I will verify this once more.